tracker-radar-tests/TR-domain-matching/domain_matching_tests.json

```diff
@@ -214,7 +214,7 @@
             },
             {
                 "name": "skip surrogates for same entity",
-                "siteURL": "https://tracker.test/",
```

Comment:
that was the bug, treacker.test and surrogates.test were not from the same entity (as test name suggests)

Reply:
Not sure I follow.  Below (https://github.com/duckduckgo/privacy-reference-tests/pull/18/files#diff-8b2239c2a9698d7409f6e31535f436f73c9d7262365409d5e6b0f4bb9bec79fdL119) `tracker.test` and `surrogates.test` are in the entity same, are they not?

Reply:
Good catch, I missed that I haven't removed `surrogates.test` from that entity. Fixed now and added check for multiple owners.

The situation before was that `tracker.test` and `surrogates.test` had the same owner according to `entities`, but different owners according to their `owner` field and `domains` object. This was the data inconsistency that I'm mentioning in the description.

The way I fixed it is by making sure that owner information in `entities` is the same as in `owner` and `domains`. In the process I decided that it will be cleaner to separate `trackers.test` and `surrogates.test` owners because that was the convention for other domains (tracker domains were under "Test Site for Tracker Blocking" entity, ignored domains were under "Ignore Site for Tracker Blocking" and I created "Test Site for Surrogates" for surrogates).